android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java

```diff
@@ -169,10 +169,15 @@ public void addTabItemInController(TiViewProxy tabProxy)
 		int index = this.mMenuItemsArray.size() - 1;
 		updateDrawablesAfterNewItem(index);
 		// Handle shift mode.
-		if (proxy.getProperties().optBoolean(TiC.PROPERTY_SHIFT_MODE, true)) {
-			this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_AUTO);
-		} else {
-			this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_UNLABELED);
+		final int shiftMode = proxy.getProperties().optInt(TiC.PROPERTY_SHIFT_MODE, 1);
+		switch (shiftMode) {
+			case 0:
+				this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_SELECTED);
+			case 1:
+				this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_AUTO);
+			case 2:
+				// NOTE: Undocumented for now, will create new property that has parity with iOS.
+				this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_UNLABELED);
```

Comment:
You are missing `break` statements. Also, `case 0` needs to be set to `LABELED` to match the old behavior which always shows tab title. Setting it to `SELECTED` means it only shows a tab title when selected which is what `case 1` is supposed to do.
```java
switch (shiftMode) {
	case 0:
		this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_LABELED);
		break;
	case 1:
		this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_AUTO);
		break;
	case 2:
		// NOTE: Undocumented for now, will create new property that has parity with iOS.
		this.mBottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_UNLABELED);
		break;
}
```